### PR TITLE
NAS-109037 / 21.02 / set comconsole,efi on UEFI booted mseries systems

### PR DESCRIPTION
**Pull request by yocalebo**
Gen3 M-series systems come with m.2 nvme boot drives. This means they're UEFI booted. The current code only enables `comconsole` if we're booted UEFI and the efi firmware provides a serial device.

If only setting `comconsole`, then the BSD boot loader doesn't appear on the convenient iKVM/HTML5 website for the m-series devices. This is undesired behavior since logging into a website as opposed to running `ipmitool sol` is subjectively easier.

This change will set the kernel console setting to `comconsole,efi` if it's UEFI booted and an m-series device.

### CODE REVIEW

**Review by william-gr — COMMENTED** [review]
Just so I understand, we want this for all M-Series, not just Gen3?

**Comment by yocalebo** [comment]
> Just so I understand, we want this for all M-Series, not just Gen3?

It should only happen for Gen3 series at the moment since the older generation m-series are booted using SATA DOMs which means they're not UEFI booted.

**Comment by william-gr** [comment]
I thought earlier Gens were not configured to boot UEFI by default, but they could

**Comment by yocalebo** [comment]
> I thought earlier Gens were not configured to boot UEFI by default, but they could

That's correct, the earlier ones are not configured to boot UEFI. However, if they do change to boot UEFI this change will apply to them as well. Which is the desired behavior.

File: src/freenas-installer/etc/install.sh [hunk 1]
```diff
@@ -392,6 +392,11 @@ videoconsole()
     fi
 }
 
+mseries()
+{
+    dmidecode -s system-product-name | grep -i "TRUENAS-M" | grep -v "MINI"
```
Comment (freqlabs) [inline src/freenas-installer/etc/install.sh:397]:
```suggestion
    dmidecode -s system-product-name | grep -iv "MINI" | grep -iq "TRUENAS-M"
```

**Review by freqlabs — APPROVED** [review]

File: src/freenas-installer/etc/install.sh [hunk 2]
```diff
@@ -392,6 +392,11 @@ videoconsole()
     fi
 }
 
+mseries()
+{
+    dmidecode -s system-product-name | grep -iv "MINI" | grep -qv "TRUENAS-M"
```
Comment (freqlabs) [inline src/freenas-installer/etc/install.sh:397]: Er, that wasn't what I suggested.

**Review by freqlabs — APPROVED** [review]

**Pull request merged**